src/libponyrt/ds/fun.c

```diff
@@ -1,8 +1,112 @@
 #include "fun.h"
+#include <stdio.h>
 #include <string.h>
 
 #include <platform.h>
 
+
+#ifdef PLATFORM_IS_ILP32
+/*
+   SipHash reference C implementation
+
+   Copyright (c) 2016 Jean-Philippe Aumasson <jeanphilippe.aumasson@gmail.com>
+
+   To the extent possible under law, the author(s) have dedicated all copyright
+   and related and neighboring rights to this software to the public domain
+   worldwide. This software is distributed without any warranty.
+
+   You should have received a copy of the CC0 Public Domain Dedication along
+   with
+   this software. If not, see
+   <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+/* default: SipHash-2-4 */
+#define cROUNDS 2
```

Comment:
To be more consistent with the 64-bit implementation, these constants can be removed, and the loops using them can be unrolled.